security-demo/README.md

```diff
@@ -0,0 +1,263 @@
+integrated with kubeege for the same.
+
+![image](./doc/images/demo-arch.png)
+
+## Source Folders
+* release : Directory has configurations and scripts to be used for deployment of identity management infrastructure for Kubeedge.
+
+* app-agent-conf  : Configurations for spire agent interfacing with edge (event-bus) and user applications . Spire agent communicates to edge spire server.|
+
+* conf : Configurations for spire server (cloud and edge) and spire agent (agent communicating to cloud spire server).
+
+* certs : Spiffe helper communicates with spire agents and download certificates in this folder . Ghostunnel is loaded with these certificates . For edge to user interface application , spiffe helper configuration and ghostunnel certs are present in event-bus and user-app folders respectively.
+
+* app-binaries : All binaries for edge , cloud and test applications.
+
+* log : Logs for different processes run as part of the example.
+
+## Scripts
+* deploy-cloud.sh : Deploys cloud part of identity management infrastructure. Used to 
+                    
+                    1) Start cloud spire server.
+                           
+                    2) Registers cloud node and cloud agent.
+                           
+                    3) Registers and starts cloud hub (test cloud hub command simulator).
+
+* deploy-edge.sh  : Deploys edge part of identity management infrastructure.
+                           
+                    1) Registers edge node and edge agent communication with cloud spire
+                           server.
+                           
+                    2) Registers and starts edge spire server.
+                           
+                    3) Registers and starts edge agent for edge to user app communication interface.
+
+* start-spiffe-helper.sh  : Based on the configuration (helper.conf) , starts spiffe-helper communication for certificate download and rotation. Ghostunnel is run using spiffe-helper.
+
+* commands.sh  : Abstracts spire cli commands.
+ 
+## Prerequisites
+* Bash version 4.4.19
+* Go compiler version 1.11.4
+* go dep and glide package managers
+* sshpass
+
+## How to configure
+### Cloud node configuration
+
+*Upstream CA – Cloud spire server configuration:*
+&lt;SPIRE\_PATH&gt;/conf/server/server.conf
+
+*Upstream CA Agent – Cloud node agent configuration :*
+&lt;SPIRE\_PATH&gt;/conf/agent/agent.conf
+
+Following fields might require modification based on the deployment environment
+
+       bind_address = "192.168.56.101" <Cloud node vm IP>
+
+       server_address = "192.168.56.101" <Cloud Spire Server IP>
+
+       server_port = "8081" <Cloud Spire Server port>
+
+       socket_path =/tmp/agent.sock"
+
+### Edge node configuration
+
+*Edge agent configuration to **connect to cloud spire server** :*
+&lt;SPIRE\_PATH&gt;/conf/agent/agent.conf
+
+Following fields might require modification based on the deployment environment
```

Comment:
Following are the fields of what configuration file?

Reply:
Also , detail configuration to be added about spiffe helper.

Reply:
Updated the details for spiffe helper.